c7n/resources/aws.py

```diff
@@ -123,6 +125,32 @@ def shape_validate(params, shape_name, service):
         raise PolicyValidationError(report.generate_report())
 
 
+def get_bucket_region_clientless(bucket):
+    """Attempt to determine a bucket region without a client
+
+    We can make an unauthenticated HTTP HEAD request to S3 in an attempt to find a bucket's
+    region. This avoids some issues with cross-account/cross-region uses of the
+    GetBucketPolicy API action.
+
+    Note that this approach will only work in the standard AWS partition.
+
+    Return a region string, or None if we're unable to determine one.
+    """
+    region = None
+    request = Request(f'https://{bucket}.s3.amazonaws.com', method='HEAD')
```

Comment:
can you ref the link where this suggestion came from?

Reply:
Sure, good call :+1: